libraries/plugins/peerplays_sidechain/peerplays_sidechain_plugin.cpp

```diff
@@ -73,23 +70,87 @@ void peerplays_sidechain_plugin::plugin_set_program_options(
    cfg.add(cli);
 }
 
-void peerplays_sidechain_plugin::plugin_initialize(const boost::program_options::variables_map& options)
+void peerplays_sidechain_plugin_impl::plugin_initialize(const boost::program_options::variables_map& options)
 {
-   ilog("peerplays sidechain plugin:  plugin_initialize()");
+   config_ready_son = options.count( "son-id" ) && options.count( "peerplays-private-key" );
+   if (config_ready_son) {
+   } else {
+      wlog("Haven't set up SON parameters");
+   }
 
-   if( options.count( "bitcoin-node-ip" ) && options.count( "bitcoin-node-zmq-port" ) && options.count( "bitcoin-node-rpc-port" )
-      && options.count( "bitcoin-node-rpc-user" ) && options.count( "bitcoin-node-rpc-password" )
-      && options.count( "bitcoin-address" ) && options.count( "bitcoin-public-key" ) && options.count( "bitcoin-private-key" ) )
-   {
-      my->_net_manager.create_handler(network::bitcoin, options);
+   net_manager = std::unique_ptr<sidechain_net_manager>(new sidechain_net_manager(plugin.app().chain_database()));
+
+   config_ready_bitcoin = options.count( "bitcoin-node-ip" ) &&
+           options.count( "bitcoin-node-zmq-port" ) && options.count( "bitcoin-node-rpc-port" ) &&
+           options.count( "bitcoin-node-rpc-user" ) && options.count( "bitcoin-node-rpc-password" ) &&
+           options.count( "bitcoin-address" ) && options.count( "bitcoin-public-key" ) && options.count( "bitcoin-private-key" );
+   if (config_ready_bitcoin) {
+      net_manager->create_handler(sidechain_type::bitcoin, options);
+      ilog("Bitcoin sidechain handler created");
    } else {
-      wlog("Haven't set up bitcoin sidechain parameters");
+      wlog("Haven't set up Bitcoin sidechain parameters");
```

Comment:
throw by now. when we get ethereum we should exit if:
- none of the 2 sidechain params are present.
- some params for bitcoin are present but not all.
- some params for ethereum are present but not all.

Reply:
Agree..